File: Backend/Database/Model/Spare.cs
using FluentValidation;
using System;

namespace Database.Model
{
    public class Spare : BaseEntity
    {
        public string Name { get; set; }
        public decimal Price { get; set; }
        public string ImageUrl { get; set; }
        public string Description { get; set; }
        public string Url { get; set; }
        public int ViewCount { get; set; } = 1;

        public Guid ModelId { get; set; }
        public virtual Model Model { get; set; }

        public Guid CategoryId { get; set; }
        public virtual Category Category { get; set; }

        public Guid ProviderId { get; set; }
        public virtual Provider Provider { get; set; }
    }

    public class SpareValidator : AbstractValidator<Spare>
    {
        public SpareValidator()
        {
            RuleFor(p => p.Name).NotEmpty();
            RuleFor(p => p.Price).NotEmpty();
            RuleFor(p => p.ImageUrl).NotEmpty();
            RuleFor(p => p.Description).NotEmpty();
            RuleFor(p => p.Url).NotEmpty();
        }
    }
}

File: Frontend/src/app/shared/models/spare.ts
import { ICategory } from './category';
import { IModel } from './model';
import { IProvider } from './provider';

export interface ISpare {
    id: number;
    name: string;
    price: number;
    imageUrl: string;
    description: string;
    url: string;
    modelId: string;
    model?: IModel;
    categoryId: string;
    category?: ICategory;
    providerId: string;
    provider?: IProvider;
    viewCount?: number;
}
